debug/src/debug.js

```diff
@@ -259,6 +263,8 @@ export function initDebug() {
 			}
 		}
 
+		if (oldDiffed) oldDiffed(vnode);
```

Comment:
Hooks use the `options.diffed` to flush `layoutEffects`. Debug needs to run its layout effect checks before the hooks option resets them to an empty array, so I moved the calling of other diffed options to after the layout effects checks are run.

This test passed previously because before each test we would delete the diffed option which deleted the hooks option and we never set it back. So the diffed hook option never ran in the old debug tests. Now it does in the new `debug-hooks.test.js` file.

Reply:
Oh that's a good catch :+1: